PWGHF/Utils/utilsMcMatching.h

```diff
@@ -0,0 +1,244 @@
+/// \return a map of final states with their corresponding PDG codes
+inline std::unordered_map<DecayChannelMain, const std::vector<int>> getDecayChannel3Prong(int pdgMother)
+{
+  switch (pdgMother) {
+    case o2::constants::physics::Pdg::kDPlus:
+      return DaughtersDPlusMain;
+    case o2::constants::physics::Pdg::kDS:
+      return DaughtersDsMain;
+    case o2::constants::physics::Pdg::kDStar:
+      return DaughtersDstarMain;
+    case o2::constants::physics::Pdg::kLambdaCPlus:
+      return DaughtersLcMain;
+    case o2::constants::physics::Pdg::kXiCPlus:
+      return DaughtersXiCMain;
+    default:
+      LOG(error) << "Unknown PDG code for 3-prong final states: " << pdgMother;
+      return {};
+  }
+}
+
+/// Returns a map of the resonant decay channels for a specific 3-prong particle specie
+/// \param pdgMother PDG code of the mother particle
+/// \return a map of resonant decay channels with their corresponding PDG codes
+inline std::unordered_map<DecayChannelResonant, const std::array<int, 2>> getResoChannels3Prong(int pdgMother)
+{
+  switch (pdgMother) {
+    case o2::constants::physics::Pdg::kDPlus:
+      return DaughtersDPlusResonant;
+    case o2::constants::physics::Pdg::kDS:
+      return DaughtersDsResonant;
+    case o2::constants::physics::Pdg::kDStar:
+      return DaughtersDstarResonant;
+    case o2::constants::physics::Pdg::kLambdaCPlus:
+      return DaughtersLcResonant;
+    case o2::constants::physics::Pdg::kXiCPlus:
+      return DaughtersXiCResonant;
+    default:
+      LOG(error) << "Unknown PDG code for 3-prong final states: " << pdgMother;
+      return {};
+  }
+}
+
+} // namespace hf_cand_3prong
+
+/// Perform the matching for a single resonant channel
+/// \tparam N size of the array of daughter PDG codes
+/// \param arrPdgResoChn array of daughter indices
+/// \param arrPdgDaugs array of PDG codes for the resonant decay
+/// \return true if the resonant channel is matched, false otherwise
+template <std::size_t N>
+inline bool checkResonantDecay(std::array<int, N> const& arrPdgResoChn, std::array<int, N> arrPdgDaugs)
+{
+  for (std::size_t i = 0; i < N; i++) {
+    bool findDaug = false;
+    for (std::size_t j = 0; j < N; j++) {
+      if (std::abs(arrPdgResoChn[i]) == std::abs(arrPdgDaugs[j])) {
+        arrPdgDaugs[j] = -1; // Mark as found
+        findDaug = true;
+        break;
+      }
+    }
+    if (!findDaug) {
+      return false;
+    }
+  }
+  return true;
+}
+
+/// Flag the resonant decays
+/// Function documentation:
+/// \tparam is3Prong bool to specify if the mother decays with a 3-prong decay
+/// \tparam N size of the array of daughter PDG codes
+/// \param motherPdg PDG code of the mother particle
+/// \param channel decay channel flag to be set
+/// \param arrDaughPdgs array of daughter PDG codes
+template <bool is3Prong = false, std::size_t N>
+inline void flagResonantDecay(int motherPdg, int8_t* channel, std::array<int, N> const& arrDaughPdgs)
```

Comment:
`channel` should be set to `0` by default since the function is supposed to always give a correct value.